src/main/java/org/edumips64/Main.java

```diff
@@ -145,9 +146,23 @@ public static void main(String args[]) {
 
     // Configure logger format.
     System.setProperty("java.util.logging.SimpleFormatter.format", "%1$tm%1$td %1$tH:%1$tM:%1$tS %4$s %2$s %5$s%6$s%n");
-
     Main.showVersion();
 
+    if (cliArgs.isHeadless()) {
+      startCli(mm.configStore, cliArgs);
+    } else {
+      checkAndStartGui(mm, cliArgs);
+    }
+  }
+
+  private static void checkAndStartGui(Main mm, Args cliArgs) {
+    if (GraphicsEnvironment.isHeadless()) {
+      log.log(Level.SEVERE, "Cannot start GUI in a headless environment.");
```

Comment:
This error message should probably be localized.

Reply:
On it!